test/PredictiveDeployer.t.sol

```diff
@@ -134,7 +167,40 @@ contract PredictiveDeployerTest is Test {
         IPredictiveDeployer(address(proxy)).deploy(invalidPrincipal, signature, childBytecode);
     }
 
-    function test_Receive(uint256 transferAmount) public {
+    function testFuzz_DeployAndUpdateNonce(uint256 pkNum) public {
```

Comment:
We can probably test this nonce update functionality in the main `testFuzz_deploy()` test function, rather than a separate test. That said, I'm open to being persuaded otherwise.